src/TestFramework/Coverage/ProxyTrace.php

```diff
@@ -108,6 +108,10 @@ public function getTests(): ?TestLocations
 
     public function getAllTestsForMutation(NodeLineRangeData $lineRange, bool $isOnFunctionSignature): iterable
     {
+        if ($this->lazyTestLocations === null) {
+            return [];
+        }
```

Comment:
`NodeMutationGenerator` does not check if there are tests, just asks for them. It is not their call to know the details, so we're doing it here.